lib/server/server.js

```diff
@@ -298,9 +318,99 @@ function convertApnToFcmToken(appName, tokenObject) {
   });
 }
 
-function getIosBundleId(vet) {
-  if (vet.appIdentifier === 'faulkvilleanimalhospital') {
-    return `com.myvet2pet.${vet.appIdentifier}`;
+function getConfigForApp(app, resolve, reject) {
+  app.getConfig()
+    .then(response => {
+      resolve(response);
+    })
+    .catch(error => {
+      reject(error);
+    });
+}
+
+function listApps(projectManagement, platform) {
+  switch (platform) {
+    case PLATFORMS.ANDROID: {
+      return projectManagement.listAndroidApps();
+    }
+    case PLATFORMS.IOS: {
+      return projectManagement.listIosApps();
+    }
+    default: {
+      throw new Error(`Invalid platform: ${platform}`);
+    }
+  }
+}
+
+function createApp(projectManagement, vet, platform) {
+  switch (platform) {
+    case PLATFORMS.ANDROID: {
+      return projectManagement.createAndroidApp(GlobalVets.VetUtils.getAndroidPackageName(vet));
+    }
+    case PLATFORMS.IOS: {
+      return projectManagement.createIosApp(GlobalVets.VetUtils.getIosBundleId(vet));
+    }
+    default: {
+      throw new Error(`Invalid platform: ${platform}`);
+    }
   }
-  return vet.iosLegacyAppIdentifier || `com.vet2pet.${vet.appIdentifier}`;
 }
+
+function configureApp(vetId, platform) {
+  return new Promise((resolve, reject) => {
+    if (!vetId) {
+      reject(new Error('Vet ID is required.'));
+      return;
+    }
+
+    const vet = GlobalVets.findOne({_id: vetId, $or: [{disabled: false}, {disabled: {$exists: false}}]});
+
+    if (!vet) {
+      reject(new Error(`Couldn't find vet for id ${vetId}`));
+      return;
+    }
+
+    if (!Push.FirebaseApps[vet.appIdentifier]) {
+      initializeFirebaseApp(vetId);
+
+      if (!Push.FirebaseApps[vet.appIdentifier]) {
+        reject(new Error(`Firebase sdk for ${vetId} is not initialized.`));
+        return;
+      }
+    }
+
+    const projectManagement = Push.FirebaseApps[vet.appIdentifier].projectManagement();
+
+    if (!projectManagement) {
+      reject(new Error(`Could not initialize FirebaseAdmin.projectManagement for ${vetId}.`));
+      return;
+    }
+
+    listApps(projectManagement, platform).then(apps => {
+
+      if (!apps || apps.length === 0) {
+        createApp(projectManagement, vet, platform)
+          .then(app => {
+            if (!app) {
+              reject(new Error(`Error creating android app for ${vetId}`));
```

Comment:
@mjovkovic Is `return` missing here?
```suggestion
              reject(new Error(`Error creating ${platform} app for ${vetId}`));
              return;
```

Reply:
@mslobodan nice catch, there should be `return` after reject.